js_modules/dagster-ui/packages/ui-core/src/assets/AutoMaterializePolicyPage/AutomaterializeMiddlePanelWithData.tsx

```diff
@@ -111,8 +111,19 @@ export const AutomaterializeMiddlePanelWithData = ({
 
   const {partitions: allPartitions} = usePartitionsForAssetKey(definition?.assetKey.path || []);
 
+  // DA will either request a list of runs or a single backfill. When DA requests a backfill
+  // we want to show a row for that backfill in the table, so we need to construct the RunsFilter
+  // differently. Backfill IDs are 8 characters long, so we can use that to determine if a
+  // backfill was requested. If DA is updated to request multiple backfills in a single evaluation,
+  // or emit a combination of runs and backfills in a single evaluation, this logic will need to
+  // be updated.
   const runsFilter: RunsFilter | null = useMemo(
-    () => (selectedEvaluation?.runIds.length ? {runIds: selectedEvaluation.runIds} : null),
+    () =>
+      selectedEvaluation?.runIds.length
+        ? selectedEvaluation.runIds.length === 1 && selectedEvaluation.runIds[0]?.length === 8
```

Comment:
why does this length need to be `8` ?

Reply:
That's the length of a backfill id

Reply:
ooooooo got it

Reply:
So wait, can't there be multiple runIds ?